volume/parser.go

```diff
@@ -0,0 +1,43 @@
+package volume
+
+import (
+	"runtime"
+
+	"github.com/docker/docker/api/types/mount"
+)
+
+const (
+	// PlatformLinux is the same as runtime.GOOS on linux
+	PlatformLinux = "linux"
+	// PlatformWindows is the same as runtime.GOOS on windows
+	PlatformWindows = "windows"
+)
+
+// Parser represents a platform specific parser for mount expressions
+type Parser interface {
```

Comment:
I was really surprised to find such a large interface, but after looking over the rest of the PR I see that this logic really is scattered all over so there's no easy way to reduce this interface. Hopefully over time we can refactor the callers and reduce the size of this interface.